src/core/dash.h

```diff
@@ -372,16 +372,15 @@ class DashTable<_Key, _Value, Policy>::Iterator {
     return *this;
   }
 
-  detail::IteratorPair<Key_t, Value_t> operator->() {
+  auto operator->() const {
     auto* seg = owner_->segment_[seg_id_];
-    return detail::IteratorPair<Key_t, Value_t>{seg->Key(bucket_id_, slot_id_),
-                                                seg->Value(bucket_id_, slot_id_)};
-  }
-
-  const detail::IteratorPair<Key_t, Value_t> operator->() const {
-    auto* seg = owner_->segment_[seg_id_];
-    return detail::IteratorPair<Key_t, Value_t>{seg->Key(bucket_id_, slot_id_),
-                                                seg->Value(bucket_id_, slot_id_)};
+    if constexpr (IsConst) {
```

Comment:
Id prefer to write using key_type = std::conditional_t<IsConst, const Key_t, Key_t>;
Id prefer to write using value_type = std::conditional_t<IsConst, const Value_t, Value_t>;
return detail::IteratorPair<key_type, value_type>{seg->Key(bucket_id_, slot_id_),
                                                  seg->Value(bucket_id_, slot_id_)};

Reply:
Why not just define an `IteratorPairType` at the top based on IsConst and then use it instead of auto?
The code then just becomes `return {seg->Key(), seg->Value()}`.

Both options discussed above introduce conditionals to the function body, when they can be just part of the interface

Reply:
i agree with @chakaz . It's not always possible to use `constexpr if` but if it is possible it's 10 times more readable.